app/concepts/cinema_hall/use_case/generate_seats.rb

```diff
@@ -4,15 +4,28 @@ module UseCase
     module CinemaHalls 
         class GenerateSeats < UseCase::Base::Base 
             def call(id)
-                big_alphabet = ("A".."Z").to_a
-                @cinema_hall = repository.find_by(id)
-                volume =  @cinema_hall.read_attribute('volume')
-                seats=1.step(volume,1).to_a
-                for i in 0..volume-1
-                      seats[i] = "#{i/10+1}#{big_alphabet[i%10]}"
+                begin
+                    cinema_hall = repository.find_by(id) 
+                    volume = cinema_hall.volume
+                    seats = self.cinema_matrix_representation(volume)
+
+                rescue ActiveRecord::RecordNotFound => e
                 end
                 return seats
             end
+
+            private
+            def cinema_matrix_representation(volume)
+                seats=(1..volume).to_a
+                seats.map.with_index do |seat, index|
+                    seats[index] = "#{index/10+1}#{alphabet[index%10]}"
```

Comment:
It looks that one method does too much :wink: 
```suggestion
                    seats[index] = generate_label(index)
```
and creates method `generate_label`.